gateway/gateway.go

```diff
@@ -49,10 +49,11 @@ import (
  */
 
 type webRequestT struct {
```

Comment:
Since the contents of struct changed drastically and don't even have request in it anymore, should we consider renaming it appropriately? @chandumlg